From 7eac868b8d621c141e100fa4afdeb15e49ab4bc8 Mon Sep 17 00:00:00 2001
From: mselva006c <mani_selvaraj@comcast.com>
Date: Tue, 10 Nov 2020 04:34:45 +0000
Subject: [PATCH] enforce stunnel to link with openssl-1.0.2o, 
         since the version of openssl(1.1.1g) from dunfell causing a cert
         validation failure

Source: COMCAST
License: GPLV2
Upstream-Status: Pending

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>
---
 configure.ac | 6 +++---
 src/Makefile.am | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index c9cc654..8ea82ad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -398,7 +398,7 @@ AC_ARG_ENABLE(libwrap,
 AC_MSG_NOTICE([**************************************** TLS])
 
 check_ssl_dir() { :
-    test -n "$1" -a -f "$1/include/openssl/ssl.h" && SSLDIR="$1"
+    test -n "$1" -a -f "$1/include/openssl-1.0.2o/openssl/ssl.h" && SSLDIR="$1"
 }
 
 iterate_ssl_dir() { :
@@ -456,7 +456,7 @@ AC_SUBST([SSLDIR])
 AC_DEFINE_UNQUOTED([SSLDIR], ["$SSLDIR"], [TLS directory])
 
 valid_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS -I$SSLDIR/include"
-valid_LIBS="$LIBS"; LIBS="$LIBS -L$SSLDIR/lib64 -L$SSLDIR/lib -lssl -lcrypto"
+valid_LIBS="$LIBS"; LIBS="$LIBS -L$SSLDIR/lib64 -L$SSLDIR/lib -lssl-1.0.2o -lcrypto-1.0.2o"
 
 if test "x$use_fips" = "xauto"; then
     AC_CHECK_FUNCS(FIPS_mode_set, [
@@ -476,7 +476,7 @@ if test -f src/stunnel.c && ! grep -q " built for $PACKAGE_STRING " src/dhparam.
         echo '#include "common.h"'
         echo '#ifndef OPENSSL_NO_DH'
         echo '#define DN_new DH_new'
-        openssl dhparam -noout -C 2048 | sed 's/static DH/DH/'
+        openssl-1.0.2o dhparam -noout -C 2048 | sed 's/static DH/DH/'
         echo '#endif /* OPENSSL_NO_DH */'
         echo "/* built for $PACKAGE_STRING */"
     ) > src/dhparam.c
diff --git a/src/Makefile.am b/src/Makefile.am
index 4c53d12..cd91b22 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -35,7 +35,7 @@ stunnel_CPPFLAGS += -DLIBDIR='"$(pkglibdir)"'
 stunnel_CPPFLAGS += -DCONFDIR='"$(sysconfdir)/stunnel"'

 # TLS library
-stunnel_LDFLAGS = -L$(SSLDIR)/lib64 -L$(SSLDIR)/lib -lssl -lcrypto
+stunnel_LDFLAGS = -L$(SSLDIR)/lib64 -L$(SSLDIR)/lib -lssl-1.0.2o -lcrypto-1.0.2o

 # stunnel3 script
 edit = sed \
